File: offer/20.表示数值的字符串.js
function isNumber(str) {
  // 去除字符串两端的空格
  str = str.trim();
  // 定义正则表达式
  const reg = /^[\+\-]?(\d+(\.\d*)?|\.\d+)([eE][\+\-]?\d+)?$/;
  // 判断是否匹配
  return reg.test(str);
}

File: offer/20.表示数值的字符串.go
import "regexp"

func isNumber(str string) bool {
	// 去除字符串两端的空格
	str = strings.TrimSpace(str)
	// 定义正则表达式
	reg := regexp.MustCompile(`^[\+\-]?(\d+(\.\d*)?|\.\d+)([eE][\+\-]?\d+)?$`)
	// 判断是否匹配
	return reg.MatchString(str)
}